service-proxy.js

```diff
@@ -172,6 +172,24 @@ function rateLimit(req, buildRes) {
     var isExitNode = self.isExitFor(req.serviceName);
     if (isExitNode) {
         self.rateLimiter.createServiceCounter(req.serviceName);
+        self.rateLimiter.createKillSwitchCounter(req.serviceName);
+    }
+
+    // apply kill switch safe guard first
+    if (isExitNode && self.rateLimiter.shouldKillSwitchService(req.serviceName)) {
```

Comment:
I’m super glad this isn’t coupled to the kill switch proper. There would have been complications with rate limit kill kill switches overriding manual kill switches. I presume that this kill switch lets traffic through when the rate drops to a reasonable level (no test for that case).